Python/pylifecycle.c

```diff
@@ -1372,17 +1372,17 @@ finalize_modules_delete_special(PyThreadState *tstate, int verbose)
         if (verbose) {
             PySys_WriteStderr("# restore sys.%s\n", name);
         }
-        PyObject *value = _PyDict_GetItemStringWithError(interp->sysdict,
-                                                         orig_name);
-        if (value == NULL) {
-            if (_PyErr_Occurred(tstate)) {
-                PyErr_WriteUnraisable(NULL);
-            }
-            value = Py_None;
+        PyObject *value;
+        if (PyDict_GetItemStringRef(interp->sysdict, orig_name, &value) < 0) {
+            PyErr_WriteUnraisable(NULL);
+            value = Py_NewRef(Py_None);
+        } else if (value == NULL) {
+            value = Py_NewRef(Py_None);
```

Comment:
You can code avoid duplication:
```suggestion
        if (PyDict_GetItemStringRef(interp->sysdict, orig_name, &value) < 0) {
            PyErr_WriteUnraisable(NULL);
        }
        if (value == NULL) {
            value = Py_NewRef(Py_None);
```

Reply:
I didn't recall if PyDict_GetItemStringRef() sets value or not on error. I checked my implementation and yes it does, to prevent undefined behavior. Ok, I changed the code.